File: app/src/main/java/org/michaelbel/moviemade/ui/modules/keywords/KeywordsRepository.java
package org.michaelbel.moviemade.ui.modules.keywords;

import org.jetbrains.annotations.NotNull;
import org.michaelbel.moviemade.BuildConfig;
import org.michaelbel.moviemade.data.entity.KeywordsResponse;
import org.michaelbel.moviemade.data.service.MoviesService;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class KeywordsRepository implements KeywordsContract.Repository {

    private MoviesService service;

    KeywordsRepository(MoviesService service) {
        this.service = service;
    }

    @NotNull
    @Override
    public Observable<KeywordsResponse> getKeywords(int movieId) {
        return service.getKeywords(movieId, BuildConfig.TMDB_API_KEY)
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread());
    }
}

File: app/src/main/java/org/michaelbel/moviemade/presentation/features/keywords/KeywordsRepository.kt
package org.michaelbel.moviemade.presentation.features.keywords

import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import org.michaelbel.moviemade.BuildConfig.TMDB_API_KEY
import org.michaelbel.moviemade.core.entity.Keyword
import org.michaelbel.moviemade.core.remote.Api

class KeywordsRepository(private val service: Api): KeywordsContract.Repository {

    override fun keywords(movieId: Int): Observable<List<Keyword>> =
        service.getKeywords(movieId, TMDB_API_KEY)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .map { it.keywords }
}